protocols/kad/src/behaviour.rs

```diff
@@ -623,6 +667,42 @@ where
                 self.add_provider.push((key, provider_peer.node_id));
                 return;
             }
+            KademliaHandlerEvent::GetValue { key, request_id } => {
+                let result = match self.records.get(&key) {
+                    Some(value) => Some((key.clone(), value.clone())),
+                    None => None,
+                };
+                let closer_peers = self.find_closest(&kbucket::Key::from(key), &source);
+
+                self.queued_events.push(NetworkBehaviourAction::SendEvent {
+                    peer_id: source,
+                    event: KademliaHandlerIn::GetValueRes {
+                        result,
+                        closer_peers,
+                        request_id,
+                    },
+                });
+            }
+            KademliaHandlerEvent::GetValueRes {
+                result,
+                closer_peers,
+                user_data,
+            } => {
+                if let Some(query) = self.active_queries.get_mut(&user_data) {
+                    if let QueryInfoInner::GetValue {
+                        key: _,
+                        results
+                    } = &mut query.target_mut().inner {
+                        if let Some(result) = result {
+                            results.push(result);
+                        }
+                    }}
```

Comment:
I think it would be good if the initial implementation already stops the query as soon as a value is returned.